src/Pages/Errors.php

```diff
@@ -4,73 +4,102 @@
-			$this->tlms_displayErrors();
-		}
+        if ($screen_id === 'toplevel_page_talentlms'
+            || $screen_id == 'talentlms_page_talentlms-setup'
+            || $screen_id == 'talentlms_page_talentlms-integrations'
+        ) {
+            $this->tlms_displayErrors();
+        }
 
-		if(!empty($this->talentlmsAdminErrors)){
-			foreach($this->talentlmsAdminErrors as $message){
-				echo '<div class="error notice is-dismissible">'.$message.'</div>';
-			}
-		}
-	}
+        if (! empty($this->talentlmsAdminErrors)) {
+            foreach ($this->talentlmsAdminErrors as $message) {
+                echo '<div class="error notice is-dismissible">' . $message . '</div>';
+            }
+        }
+    }
 
-	function tlms_displayErrors(){
-		if((!get_option('tlms-domain') && !get_option('tlms-apikey')) && (empty($_POST['tlms-domain']) && empty($_POST['tlms-apikey']))){
-			$this->tlms_logError('<p><strong>'.__('You need to specify a TalentLMS domain and a TalentLMS API key.', 'talentlms').'</strong>'.sprintf(__('You must <a href="%1$s">enter your domain and API key</a> for it to work.', 'talentlms'), 'admin.php?page=talentlms-setup').'</p>');
-		}
-		else{
-			try{
-				TalentLMS::setDomain(get_option('tlms-domain'));
-				TalentLMS::setApiKey(get_option('tlms-apikey'));
+    public function tlms_displayErrors(): void
+    {
+        if ((
+                empty($_POST['tlms-domain'])
+                && empty($_POST['tlms-apikey'])
+            )
+            &&
+            (
+                ! get_option('tlms-domain')
+                && ! get_option('tlms-apikey')
+            )
+        ) {
+            $this->tlms_logError(
+                '<p><strong>'
+                . esc_html__('You need to specify a TalentLMS domain and a TalentLMS API key.', 'talentlms')
+                . '</strong>'
+                . sprintf(
+                    esc_html__('You must <a href="%1$s">enter your domain and API key</a> for it to work.', 'talentlms'),
+                    esc_url(admin_url('admin.php?page=talentlms-setup'))
+                )
+                . '</p>'
+            );
+        } else {
+            try {
+                TalentLMS::setDomain(esc_html(get_option('tlms-domain')));
+                TalentLMS::setApiKey(esc_html(get_option('tlms-apikey')));
 
-				if(is_admin() && !wp_doing_ajax()){
-					Utils::tlms_getCourses();
-					Utils::tlms_getCategories();
-				}
-			}
-			catch(Exception $e){
-				if ($e instanceof TalentLMS_ApiError) {
-					$this->tlms_logError($e->getMessage());
-				}
-			}
-		}
-	}
-}
\ No newline at end of file
+                if (is_admin() && ! wp_doing_ajax()) {
+                    Utils::tlms_getCourses();
+                    Utils::tlms_getCategories();
+                }
+            } catch (Exception $e) {
+                if ($e instanceof TalentLMS_ApiError) {
+                    $this->tlms_logError($e->getMessage());
+                }
+            }
```

Comment:
We can replace this lines of code by using $this->enableTalentLMSLib();

Reply:
Please check [3f4ce03](https://github.com/epignosis/talentlms-wordpress/pull/26/commits/3f4ce0306a88aba162ffb790ce44ddbaad812e6b)